storage/storage-interface/src/state_view.rs

```diff
@@ -69,14 +68,6 @@ pub trait DbStateViewAtVersion {
 
 impl DbStateViewAtVersion for Arc<dyn DbReader> {
     fn state_view_at_version(&self, version: Option<Version>) -> Result<DbStateView> {
-        if let Some(some_version) = version {
```

Comment:
@msmouse - FYI, there is some inconsistency in this `self.get_latest_version()` is most likely stale in some cases and there were few test cases failing because of this check. In particular, `get_latest_transaction_info_option` returns a version which is higher than `get_latest_version`. So the `fetch_latest_synced_version`  function fails. This is something we might need to follow up - I will file a follow up issue for this.

Reply:
The problem is actually that the "synced version" can (reasonably) be ahead of the "latest version", which is a "committed version", which means there's a ledger info at or beyond that version. --- it's the correct thing to do to remove the check.